Chat UIKit Guide

howawong opened the pull request:
Initial version of Chat UIKit guide.

Sections included
- Installation
- XML attributes in Android & UICustomization in iOS
- Adapters in android and delegates in iOS

howawong committed:
Add UIKit

howawong committed:
Add Skygear Chat UIKit Android Guide First Draft

howawong committed:
Add Skygear Chat UIKit iOS Guide First Draft and spelling correction

rickmak commented on chat/uikit/android.md:
```diff
@@ -0,0 +1,246 @@
+7. Text Input
+8. Voice Button
+
+![ConversationView Overview](/assets/chat/uikit/android/conversation-view-overview.png)
```
The image is missing.

howawong replied:
Added in https://github.com/SkygearIO/skygear-doc/pull/552

rickmak reviewed (changes requested):
Android part Customization flow is 

Specifying XML resource -> Attribute Customization -> 
-> Event Hooks

And than go back to `Title Option` -> `Adapter` and `Avatar`

I think the event handling should not inline with appearance? They should belong to different sectioin.

tenwy commented on chat/uikit/android.md:
```diff
@@ -0,0 +1,246 @@
+
+## Components
+### Conversation
+UIKit provides the `ConversationActivity`, the main UI of a conversation.
```
I think this part (together with the image) should serve as an introduction on the top on this guide. As user, I would like to know why the UI kit is relevant to me before learning about how to install it.

tenwy commented on chat/uikit/android.md:
```diff
@@ -0,0 +1,246 @@
+#### Attribute Customization
+The following XML attributes are supported.
+
+##### Avatar
```
Please use article before a noun. e.g. background color of the avatar

tenwy commented on chat/uikit/android.md:
```diff
@@ -0,0 +1,246 @@
+|`messageStatusShouldShow`|Show message status if true|Boolean, default: True|
+
+
+#### Event Hooks
```
I think this title should be more prominent. Can we use h2 instead? Especially h2 title will appear in TOC it is good to let user know that we provide event hooks right away after looking at the TOC.

tenwy commented on chat/uikit/android.md:
```diff
@@ -0,0 +1,246 @@
+3. implementing interface and adapters.
+
+
+#### Specifying XML resource
```
I think after a H2, we should use a H3 instead to keep the consistency.

tenwy commented on chat/uikit/android.md:
```diff
@@ -0,0 +1,246 @@
+|`ConnectionListenerIntentKey`|ConnectionListener|
+
+
+#### Title Option
```
Not sure if I got it correctly, but if title option has nothing to do with event hooks, could we move it under the "Attribute Customisation" section? So that I don't have to jump between sections to understand how to customise the title of the conversation while I want to learn about how to customise the avatar for example.

tenwy commented on chat/uikit/android.md:
````diff
@@ -0,0 +1,246 @@
+    }
+};
+```
+### Avatar
````
Should this be included in the attribute customisation > avatar? So I don't have to scroll down to bottom in order to learn about everything about avatar.

howawong replied:
I think we better keep avatar away from attribute customisation because Avatar section would focuses on customisable code/interface while attribute customisation focuses on setting things in intent or XML. So I would rename it as *Avatar Adapter* instead.

tenwy reviewed (commented):
Hello @howawong I have left my comments for the Android guide. Mostly related to the organization of the doc and its readability. As I am not familiar with the UI Kits, my comments may be invalid. If so please do not be hesitate to let me know. Plus grab me if you need any clarification. Thanks!

howawong committed:
Update Android UIKit based on @tenwy 's comment #165

- Add articles
- Changed header to improve TOC
- Move title option into arttibute customization section
- Rename section "Avatar" to section "Avatar Adatper"

howawong committed:
Update chat UIKit guide iOS version with objective-C code

refs #165
